### PR TITLE
Defined `append`s behavior on some corner cases and added tests

### LINKED ISSUE #1927: `append` fails when adding nulls to FrameColumn
To reproduce

```kt
val df = dataFrameOf("col", columnOf(emptyDataFrame()))
df.append(null) // Can not add value 'null' to FrameColumn
```

For consistency with `update`, this could also turn into a `ValueColumn<DataFrame<>?>`

### LINKED ISSUE #1952: Add Unit tests for `append`
## Test scenarios for `DataFrame.append`

Based on `core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt` and the site docs: https://kotlin.github.io/dataframe/append.html

### Basic append

- Given a dataframe with columns `name`, `age`
- When calling `df.append("Bill", 30)`
- Then the result contains one additional row
- And values are assigned by column order: `name = "Bill"`, `age = 30`

### Append multiple rows

- Given a dataframe with columns `name`, `age`
- When calling `df.append("Mike", 15, "John", 17, "Bill", 30)`
- Then three rows are appended
- And values are interpreted row-wise, not column-wise

### Invalid number of values

- Given a dataframe with `n` columns
- When calling `append` with a number of values that is not a multiple of `n`
- Then an exception is thrown
- And the error explains that a multiple of the column count is expected

### Append zero rows

- Given a dataframe with at least one column
- When calling `df.append()` with no values
- Then the result has the same row count and values as the original dataframe

### Append nullable values

- Given a dataframe with nullable-compatible columns
- When calling `df.append(null, someValue)`
- Then `null` is appended as a value in the corresponding column

### Append with compiler plugin overload

- Given a `@DataSchema` class, as shown on the site
- When the compiler plugin overload is used, e.g. `df.append(Person("Bill", 30))`
- Then the row is appended according to the schema
- This should be covered in plugin/integration tests if the overload is generated there

### Empty dataframe edge case

- Given a dataframe with zero columns
- When calling `append(...)`
- Then behavior should be explicitly tested/documented, because current implementation checks `values.size % ncol`, which is problematic when `ncol == 0`

### PR DESCRIPTION
**Pull request by Allex-Nik**
Fixes #1952
Fixes #1927 

- Currently `append` fails when one appends a `null` to a `FrameColumn`. Changes from this PR make `append` add an empty frame instead of `null`, keeping the type of the column `FrameColumn`.
- Make `append` return the original dataframe early when `append` does not have any arguments.
- Make `append` with arguments fail with an informative message when the receiver dataframe has no columns.
- Add tests for `append`

### CODE REVIEW

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        val result = df.append(null)
+
+        result.nrow shouldBe 2
+        result["col"].kind() shouldBe ColumnKind.Frame
+        result["col"].type() shouldBe typeOf<DataFrame<*>>()
+        result["col"].values() shouldBe listOf(frame, DataFrame.empty(frame.schema()))
+    }
+
+    @Test
+    fun `append uses the DataRowSchema overload`() {
```
Comment (Allex-Nik) [inline core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt]: I am a bit unsure regarding this scenario. The [issue](https://github.com/Kotlin/dataframe/issues/1952) mentioned the scenario "Append with compiler plugin overload". The compiler plugin does not work in `core/...test/` but this overload from `DataRowSchemaApi.kt` seems to do what is expected (`df.append(Person("Bill", 30))`). And I also saw that there is a test checking this behavior on the compiler plugin side (`dataRowSchemaApi.kt`). Do you think it still makes sense to add a test to a module with compiler plugin working (or maybe move all these tests there)?

Image

Comment (koperagen) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt]: It does because  Person manually adds required interfaces, same what plugin does:
private data class Person(val name: String, val age: Int) : DataRowSchema
But yeah, indeed, it's a workaround and makes more sense to move this case to samples module
Not all tests

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    fun `append null to frame column replaces null with empty dataframe with schema`() {
+        val frame = dataFrameOf("value")(1)
+        val df = dataFrameOf(columnOf(frame) named "col")
+
+        val result = df.append(null)
+
+        result.nrow shouldBe 2
+        result["col"].kind() shouldBe ColumnKind.Frame
+        result["col"].type() shouldBe typeOf<DataFrame<*>>()
+        result["col"].values() shouldBe listOf(frame, DataFrame.empty(frame.schema()))
```
Comment (koperagen) [inline core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt]: Please add something like this too
```
result["col"].values().checkAll { 
  shouldNotThrowAny { 
    it["value"] // check column access
  }
}
```

**Review by koperagen — COMMENTED** [review]
Nice :)

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt [hunk 3]
```diff
@@ -9,8 +9,13 @@ import org.jetbrains.kotlinx.dataframe.nrow
 // region DataFrame
 
 public fun <T> DataFrame<T>.append(vararg values: Any?): DataFrame<T> {
```
[the remark this answers is not in the document]
Comment (Allex-Nik) [reply core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt:11]: I am planning to do it in the next PR fixing #1950 and #1951

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt [hunk 4]
```diff
@@ -128,10 +128,14 @@ internal fun <T, C> DataColumn<C>.updateImpl(
 internal fun <T> DataColumn<T>.updateWith(values: List<T>): DataColumn<T> =
```
[the remark this answers is not in the document]
Comment (Allex-Nik) [reply core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt:131]: Done

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt (hunk 1 above)
Comment (Allex-Nik) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt]: Done

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt (hunk 2 above)
Comment (Allex-Nik) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/append.kt]: Done

**Review by koperagen — APPROVED** [review]

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt [hunk 5]
```diff
@@ -123,15 +123,21 @@ internal fun <T, C> DataColumn<C>.updateImpl(
...
+            val frameSchema by schema
+            val groups = values.map {
+                when (it) {
+                    null -> DataFrame.empty(frameSchema)
+                    is AnyFrame -> it
+                    else -> throw IllegalArgumentException("Can not add value '$it' to FrameColumn")
+                }
             }
-            val groups = (values as List<AnyFrame>)
             DataColumn.createFrameColumn(name, groups) as DataColumn<T>
```
[the remark this answers is not in the document]
Comment (Allex-Nik) [reply core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt]: Done
